projects/testing/src/matchers/matchers.ts

```diff
@@ -621,6 +899,18 @@ export interface SkyMatchers<T> extends jasmine.Matchers<T> {
    * @param callback The callback to execute when the comparison fails.
    */
   toMatchResourceTemplate(name: string, callback?: () => void): void;
+
+  /**
+   * `expect` the actual element to have the text for the expected resource string.
+   * Uses `SkyLibResourcesService.getString(name, args)` to fetch the expected resource string
+   * and compares the tokenized element text against the template.
+   * Essentially this matches any text that has the non-parameterized text of the template in the order of the template,
+   * regardless of the value of each of the parameters.
+   * @deprecated Use `await expectAsync(element).toMatchResourceTemplate('foo_bar_key')` instead.
```

Comment:
For net-new stuff, we should probably just go with async matchers (and not introduce another thing we need to deprecate/remove in the future).